src/main/java/com/aws/greengrass/componentmanager/builtins/S3Downloader.java

```diff
@@ -153,14 +173,22 @@
         }
     }
 
-    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.AvoidRethrowingException"})
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.AvoidRethrowingException", "PMD.PreserveStackTrace"})
     private S3Client getRegionClientForBucket(String bucket) throws InterruptedException, PackageDownloadException {
         GetBucketLocationRequest getBucketLocationRequest = GetBucketLocationRequest.builder().bucket(bucket).build();
         String region = null;
         try {
-            region = RetryUtils.runWithRetry(s3ClientExceptionRetryConfig,
-                    () -> s3ClientFactory.getS3Client().getBucketLocation(getBucketLocationRequest)
-                            .locationConstraintAsString(), "get-bucket-location", logger);
+            region = RetryUtils.runWithRetry(s3ClientExceptionRetryConfig, () -> {
+                try {
+                    return s3ClientFactory.getS3Client().getBucketLocation(getBucketLocationRequest)
+                            .locationConstraintAsString();
+                } catch (S3Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)